types/ws/index.d.ts

```diff
@@ -346,6 +346,9 @@ declare namespace WebSocket {
         removeListener(event: string | symbol, listener: (...args: any[]) => void): this;
     }
 
+    class WebSocketServer extends Server {}
+    class WebSocket extends WebSocketAlias {}
```

Comment:
```suggestion
    export WebSocket = WebSocketAlias;
```

Reply:
Your suggestion doesn't work well.

```diff
diff --git a/types/ws/index.d.ts b/types/ws/index.d.ts
index 876496d06f..3760cf3d91 100644
--- a/types/ws/index.d.ts
+++ b/types/ws/index.d.ts
@@ -171,7 +171,7 @@ declare class WebSocket extends EventEmitter {
     removeListener(event: string | symbol, listener: (...args: any[]) => void): this;
 }
 
-declare class WebSocketAlias extends WebSocket {}
+import WebSocketAlias = WebSocket;
 
 declare namespace WebSocket {
     /**
@@ -347,7 +347,7 @@ declare namespace WebSocket {
     }
 
     class WebSocketServer extends Server {}
-    class WebSocket extends WebSocketAlias {}
+    export type WebSocket = WebSocketAlias;
 
     // WebSocket stream
     function createWebSocketStream(websocket: WebSocket, options?: DuplexOptions): Duplex;
```

```
$  npm test ws

> definitely-typed@0.0.3 test /Users/tamura/src/github/DefinitelyTyped
> dtslint types "ws"

Error: /Users/tamura/src/github/DefinitelyTyped/types/ws/index.d.ts:350:5
ERROR: 350:5  strict-export-declare-modifiers  'export' keyword is redundant here because all declarations in this module are exported automatically. If you have a good reason to export some declarations and not others, add 'export {}' to the module to shut off automatic exporting. See: https://github.com/Microsoft/dtslint/blob/master/docs/strict-export-declare-modifiers.md

    at /Users/tamura/src/github/DefinitelyTyped/node_modules/dtslint/bin/index.js:207:19
    at Generator.next (<anonymous>)
    at fulfilled (/Users/tamura/src/github/DefinitelyTyped/node_modules/dtslint/bin/index.js:6:58)
npm ERR! Test failed.  See above for more details.
```

Reply:
Your suggestion doesn't work well.

VS Code says the following line is invalid:
```ts
export WebSocket = WebSocketAlias;
```

Reply:
@rbuckton I had better make changes as suggested even if VS Code says it is invalid?

Reply:
@rbuckton is there any down sides to @tamuratak's version?

Reply:
@rbuckton ping

Reply:
@rbuckton's code is missing a `const` or the like, eg
```ts
export const WebSocket = WebSocketAlias;
```
but that won't also reexport the types - you'd need
```ts
export {WebSocketAlias as WebSocket};
```
for that, ideally, but that's not allowed within a namespace declaration, so the best you could do is probably
```ts
export const WebSocket = WebSocketAlias;
export type WebSocket = WebSocketAlias;
```
to export both meanings. It's probably better than the class redeclarations that imply a prototype hierarchy that doesn't exist.